sungc/deglint.py

```diff
@@ -940,88 +1398,82 @@ def hedley_2005(
-                valid_nir = nir_im[roi_valix]
-                min_refl_nir = valid_nir.min()
-
-                # 2. Get correlations between current band and NIR
-                y_vals = vis_im[roi_valix]
-                slope, y_inter, r_val, p_val, std_err = stats.linregress(
-                    x=valid_nir, y=y_vals
-                )
-
-                # 3. deglint water pixels
-                deglint_band[water_mask] = vis_im[water_mask] - slope * (
-                    nir_im[water_mask] - min_refl_nir
+                if plot:
+                    vis_bname = f"B{res_ordered_vis[res][bname][0]}"
+                    plot_tuple = (fig, ax, vis_bname, f"B{corr_band}", odir)
+                else:
+                    plot_tuple = None
+
+                deglint_band, success = hedley_backend(
+                    vis_im=vis_im,
+                    corr_im=corr_im,
+                    water_mask=water_mask,
+                    roi_mask=roi_mask,
+                    nodata=nodata,
+                    scale_factor=self.scale_factor,
+                    clip=False,
+                    plot=plot,
+                    plot_tuple=plot_tuple,
                 )
-                deglint_band[(vis_im == nodata) | (nir_im == nodata)] = nodata
 
-                # 4. add 3D array (1, nrows, ncols) to xarray dict.
-                data_varname = "{0}_{1}_hedley_deglint".format(self.sub_product, bname)
-                xr_dvars[data_varname] = (
-                    ["time", "y", "x"],
-                    deglint_band.reshape(1, vis_meta["height"], vis_meta["width"]),
+                # add 3D array (1, nrows, ncols) to xarray dict.
+                data_varname = f"{self.sub_product}_{bname}"
+                xr_dvars[data_varname] = xr.Variable(
+                    dims=["time", "y", "x"],
+                    data=deglint_band.reshape(1, vis_meta["height"], vis_meta["width"]),
                 )
 
             # end-for i, bname
             dxr = create_xr(
                 xr_dvars,
                 vis_meta,
                 self.overpass_datetime,
-                f"deglinted {self.sub_product} bands {res} via Cox and Munk (1954)",
+                "Hedley et al. (2005)",
             )
 
             dxr_list.append(dxr)
 
-            # ------------------------------ #
-            if plot:
-                # create a density plot
-                plot_correlations(
-                    fig=fig,
-                    ax=ax,
-                    r2=r_val ** 2,
-                    slope=slope,
-                    y_inter=y_inter,
-                    nir_vals=valid_nir,
-                    vis_vals=y_vals,
-                    scale_factor=self.scale_factor,
-                    nir_band_id=corr_band,
-                    vis_band_id=res_ordered_vis[res][bname][0],
-                    odir=odir,
-                )
+        # cleanup
+        if isinstance(tmpdir, tempfile.TemporaryDirectory):
```

Comment:
Similar here, I'd simplify the check

```suggestion
        if not (tmpdir is None):
```